UMN-OSDFrontEnd/FrontEndTab.cs

```diff
@@ -0,0 +1,373 @@
+            groupBox.Content = stackPanel;
+
+            return stackPanel;
+        }
+
+        public TextBlock AddTextBlock(string text)
+        {
+            TextBlock textBlock = new TextBlock
+            {
+                HorizontalAlignment = HorizontalAlignment.Left,
+                Margin = new Thickness(5, 5, 5, 5),
+                TextWrapping = TextWrapping.Wrap,
+                VerticalAlignment = VerticalAlignment.Top,
+                Text = text
+            };
+
+            MainPanel.Children.Add(textBlock);
+
+            return textBlock;
+        }
+
+        public void AddCheckBoxGroup(List<FrontEndCheckBox> frontEndCheckBoxes, StackPanel stackPanel)
+        {
+            _ = frontEndCheckBoxes ?? throw new ArgumentNullException(nameof(frontEndCheckBoxes));
+
+            foreach (FrontEndCheckBox frontEndCheckBox in frontEndCheckBoxes)
+            {
+                frontEndCheckBox.Style = CheckBoxStyle;
+                stackPanel.Children.Add(frontEndCheckBox);
+            }
+        }
+
+        public void AddDropDownGroup(string tsVariable, List<string> options, StackPanel stackPanel, string defaultSelectedValue = null)
+        {
+            FrontEndComboBox comboBox = new FrontEndComboBox
+            {
+                TSVariable = tsVariable,
+                Style = ComboBoxStyle,
+                SelectedIndex = 0
+            };
+
+            foreach (string option in options)
+            {
+                ComboBoxItem comboBoxItem = new ComboBoxItem
+                {
+                    Content = option,
+                    Style = ComboBoxItemStyle
+                };
+
+                comboBox.Items.Add(comboBoxItem);
+            }
+
+            if (defaultSelectedValue != null)
+            {
+                int selectionIndex = -1;
+                foreach (ComboBoxItem comboBoxItem in comboBox.Items)
+                {
+                    if (comboBoxItem.Content.ToString() == defaultSelectedValue)
+                    {
+                        selectionIndex = comboBox.Items.IndexOf(comboBoxItem);
+                    }
+                }
+
+                if (selectionIndex > -1)
+                {
+                    comboBox.SelectedIndex = selectionIndex;
+                }
+                else
+                {
+                    if(Development)
+                    {
+                        MessageBox.Show("Didn't find index of " + defaultSelectedValue);
+                    }
+                }
+            }
+
+            stackPanel.Children.Add(comboBox);
+        }
+
+        public void ProcessTabLayout(List<TabLayout> tabLayout)
+        {
+            foreach (TabLayout tabLayoutItem in tabLayout)
+            {
+                switch (tabLayoutItem.ItemType)
+                {
+                    case "textBlock":
+                        AddTextBlock(tabLayoutItem.ManualTextBlock.Text);
+                        break;
```

Comment:
This can't handle the case of a dynamic text block that's meant to have the text from a ts variable per the example appsetting json. Perhaps a similar check to the rest of the cases might be needed 
`if(dynamic) {
    if(!development) {
          assign some default value and addtextblock
}
else {
      grab text from specified ts variable and addtextblock
}
}
else {
 addtextblock....
}`